sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -10,3 +10,175 @@
 
 #include <_az_cfg.h>
 
+static const az_span provisioning_service_api_version
+    = AZ_SPAN_LITERAL_FROM_STR("/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION);
```

Comment:
does this not need a `?`

Reply:
This works today.
Where do you think this needs the '?' - prefix or postfix?

Reply:
I only ask because IoT hub seems to be inconsistent where they are needed or not.
On connect, needed: [LINK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#using-the-mqtt-protocol-directly-as-a-device)
On telemetry, not needed: [LINK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#sending-device-to-cloud-messages)
On twin, needed: [LINK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#retrieving-a-device-twins-properties)

Reply:
Good point: could you please try the other features w/o the `?`?
I'll leave this one w/o the `?` because I was able to get the server to respond and, apparently, to the point where I got an operationId everything works fine. If we find issues, we can add the ? later.

Reply:
On hub it seems to work without the `?`. At least from using MQTT Explorer. As long as it works totally good with me 👍